addons/web/static/src/js/core/custom_hooks.js

```diff
@@ -36,8 +36,51 @@ odoo.define('web.custom_hooks', function () {
             component.__willFocus = selector;
         };
     }
+    /**
+     * When a component needs to set up generic event handling on its root node
+     * and makes it available for all subclasses
+     * Useful in many parts of Odoo, when a subclass of a component
+     * usually *adds* behavior relatively to its parent class
+     *
+     * Example: navigation event handling in Fields
+     *
+     * Usage:
+     *   In the constructor of the OWL component
+     *
+     * Note:
+     *   When selector contains a native querySelector, event delegation is activated
+     *   Components that alter Event.target (e.g. Portal) are not expected
+     *   to behave as expected in that setting
+     *
+     * @param {String} selector. A string containing the event name, a space
+     *     and then a valid native JS selector
+     * @param {Function}. The handler of the event; will be bound to component
+     */
+    function useListener(selector, handler) {
```

Comment:
I don't like the name of arg 'selector'. We should find a better name (I have no idea right now), or split the arg into two: 'eventName' and 'selector'. In the second case, it would imply a function with multiples APIs (which is something I don't recommend in general):
```
useListener(eventName, handler)
useListener(eventName, selector, handler)
```